### PR TITLE
Re-implement models in our AISTATS paper and more

### PR DESCRIPTION
**Pull request by xukai92**
AISTATS paper

- [x] High-dim Gaussian
- [x] Naive Bayes
- [x] HMM

Extra

- [x] Logistic regression
- [x] Stochastic volatility
- [x] Gaussian with unknown parameters
- [x] Hierarchical Poisson
- [x] LDA

### CODE REVIEW

File: aistats-2018/naive-bayes/model.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using Turing
+
+Turing.setadbackend(:reverse_diff)
+
+@model naive_bayes(image, label, C, ::Type{T}=Float64) where {T<:Real} = begin
+    D, N = size(image)
+    
+    m = Matrix{T}(undef, D, C)
+    for c = 1:C
```
Comment (xukai92) [inline aistats-2018/naive-bayes/model.jl]: @mohamed82008 I tried `@. m ~ Normal(0, 1)` to replace this loop but the inference result differs. Did I do something wrong or it indicates there might be a bug in our broadcast operation?

File: aistats-2018/naive-bayes/model.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        m[:,c] ~ MvNormal(fill(0.0, D_pca), fill(1.0, D_pca))
+    end
+
+    Threads.@threads for n = 1:N
```
Comment (xukai92) [inline aistats-2018/naive-bayes/model.jl]: (Rought) inference time
- 1 threads: ~ 1 hour
- 10 threads: 10 mins

Comment (xukai92) [reply aistats-2018/naive-bayes/model.jl]: @trappmartin Maybe we should add this to our performance tips.

Comment (trappmartin) [reply aistats-2018/naive-bayes/model.jl]: Cool, I'll add this to the performance tips.

File: aistats-2018/naive-bayes/model.jl (hunk 1 above)
Comment (mohamed82008) [reply aistats-2018/naive-bayes/model.jl]: I think there is a bug there. I will look into it this weekend.

**Comment by xukai92** [comment]
It's ready to merge. I do this after the abstract sub.

**Pull request merged**